src/Microsoft.TemplateEngine.Utils/InMemoryFileSystem.cs

```diff
@@ -848,5 +848,40 @@ public DateTime GetLastWriteTimeUtc(string file)
 
             return targetFile.LastWriteTimeUtc;
         }
+
+        public void SetLastWriteTimeUtc(string file, DateTime lastWriteTimeUtc)
+        {
+            if (!IsPathInCone(file, out string processedPath))
+            {
+                if (_basis is IFileLastWriteTimeSource lastWriteTimeSource)
+                    lastWriteTimeSource.SetLastWriteTimeUtc(file, lastWriteTimeUtc);
+                throw new NotImplementedException("Basis file system must implement IFileLastWriteTimeSource");
```

Comment:
minor - if the `IFileLastWriteTimeSource` type's name changes, this message would be outdated.  Does it make sense to cache a corresponding Type name in the instance & use it as a variable here?

Reply:
My goal here was to maintain as much similarity to existing code as possible, so `SetLastWriteTimeUtc` was essentially copied from `GetLastWriteTimeUtc` which already had the type name as part of the string.  Rather than caching a type name, I went ahead and used `nameof` instead.